Task_2/src/Pages/root.tsx

```diff
@@ -1,12 +1,87 @@
 import { Outlet } from "react-router-dom";
 import React from "react";
 import Navbar from "../components/Navbar/Navbar";
+import CartContext from "../Store/ShoppingCart";
+import AuthenticationContext from "../Store/Authentication";
+import { useState } from "react";
+
 const RootLayout: React.FC = () => {
+  const [cart, setCart] = useState<{ items: { id: number; count: number }[] }>({
+    items: [],
+  });
+
+  const [authData, setAuthData] = useState<{
+    isAuthenticated: string | null;
+    token: string | null;
+  }>({
+    isAuthenticated: localStorage.getItem("isAuthenticated"),
+    token: localStorage.getItem("token"),
+  });
+
+  const addItemToCartHandler = (id: number) => {
+    const { items } = cart;
+    let flag = true;
+    items.filter((item) => {
+      if (id === item.id) {
```

Comment:
@codal-uchoudhary there is no need of using if inside filter method. _**items.filter((item) => item.id === id)**_ directly write this. It's an incorrect syntax